tutorials/37-offline.md

```diff
@@ -6,30 +6,34 @@ Enketo may be [configured](./tutorial-10-configure.html#offline-enabled) to supp
+The Enketo app registers a [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API), which intercepts some network requests in order to [cache](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage) certain resources as they're requested. The following resources are stored in one of two caches:
 
--   The Service Worker script itself
--   The Enketo app's static assests (core JavaScript and CSS)
--   CSS supplied by [configured themes](./tutorial-10-configure.html#themes-supported)
--   The Enketo app icon, which may be displayed if a user bookmarks the app or registers it as a mobile web app
--   The initial HTML which begins loading the app and the requested form
+-   Common static assets used by Enketo to serve and render forms:
 
-This cache is cleared and rebuilt, after a brief delay, when:
+    -   The Service Worker script itself
+    -   The Enketo app's static assests (core JavaScript and CSS)
+    -   CSS supplied by [configured themes](./tutorial-10-configure.html#themes-supported)
+    -   The Enketo app icon, which may be displayed if a user bookmarks the app or registers it as a mobile web app
+    -   The initial HTML which begins loading the app and the requested form
 
--   An Enketo deployment is updated to a new version, and the user again accesses a form in offline-capable mode
--   When any change is made to the Service Worker script, configuration, or the version specified in `package.json`
+    This cache is cleared and rebuilt when an Enketo deployment is updated to a new version, and the user again accesses a form in offline-capable mode.
 
-Apart from those cases, onced cached these resources will always be retrieved from Cache Storage without performing a network request.
+-   Resources associated with individual forms:
 
-**Important note:** after such an update, previously cached forms must be requested in offline-capable mode _while online_ to be re-added to the cache.
+    -   Media attachments used by the form
+    -   [External secondary instances](https://getodk.github.io/xforms-spec/#secondary-instances---external) used by the form
 
-#### Form and form attachments
+    The resources in this cache will be updated when Enketo detects that they have changed.
+
+For production Enketo deployments, these resources will always be retrieved from the cache if present, without performing additional network requests other than to determine whether they have been updated. To aid maintenance and improvement of Enketo's offline functionality, the requests _are_ performed in development mode.
+
+**Important note:** forms cached prior to release of [thesse changes to media and HTML caching](https://github.com/enketo/enketo-express/pull/465) must be requested in offline-capable mode while online to be re-added to the cache. Forms cached after that update will no longer have this issue.
+
+#### Forms
```

Comment:
Should this heading be `IndexedDB` to parallel "Service Worker and Cache Storage"?